README.md

```diff
@@ -23,67 +23,157 @@ For a Getting started guide, API docs, recipes, making a plugin, etc. see the [d
+        }))
+    .pipe(gulp.dest(paths.styles.dest));
+}
+
+function scripts() {
+  return gulp.src(paths.scripts.src)
+    .pipe(uglify())
+    .pipe(concat(main.min.js))
+    .pipe(gulp.dest(paths.scripts.dest));
+}
+
+function watch() {
+  gulp.watch(paths.scripts.src, scripts);
+  gulp.watch(paths.styles.src, styles);
+}
+
+/* 
+ * Register some tasks to expose to the cli
+ */
 gulp.task(clean);
 gulp.task(watch);
 
-// The default task (called when you run `gulp` from cli)
-gulp.task('default', gulp.series('build'));
+/* 
+ * Notice that you can specify if tasks can run in parallel or not
+ * using `gulp.series` and `gulp.parallel`
+ */
+gulp.task('build',
+  gulp.series(clean),
+  gulp.parallel(styles, scripts));
+
+/*
+ * Expose default task that can be called by just running `gulp` from cli
+ */
+gulp.task('default',
+  gulp.series(build))
+```
+
+## Use last JavaScript version in your gulpfile
 
+Node already supports a lot of **ES2015**, to avoid compatibility problem we suggest to install Babel and rename your `gulpfile.js` as `gulpfile.babel.js`.
 
-/* Define our tasks using plain functions */
+```sh
+npm install --save-dev babel-core -babel-preset-es2015
+```
+
+Then create a **.babelrc** file with the rpeset configuration.
+
+```js
+{
+  "presets": [ "es2015" ]
+}
+```
+
+And here's the same sample from above written in **ES2015**.
+
+```js
+import gulp from 'gulp';
+import rename from 'gulp-rename';
+import less from 'gulp-less';
+import cssmin from 'gulp-cssmin';
+import concat from 'gulp-concat';
+import uglify from 'gulp-uglify';
+import del from 'del';
+
+const paths = {
+  styles: {
+    src: 'src/styles/**/*.less',
+    dest: 'assets/styles/'
+  },
+  scripts: {
+    src: 'src/scripts/**/*.js',
+    dest: 'assets/scripts/'
+  }
+};
 
-// Not all tasks need to use streams
-// A gulpfile is just another node program and you can use all packages available on npm
-// But it must return either a Promise or Stream or take a Callback and call it
 function clean() {
-  // You can use multiple globbing patterns as you would with `gulp.src`
-  // If you are using del 2.0 or above, return its promise
-  return del(['build']);
+  del([ 'assets' ]);
 }
 
-// Copy all static images
-function images() {
-  return gulp.src(paths.images)
-    // Pass in options to the task
-    .pipe(imagemin({optimizationLevel: 5}))
-    .pipe(gulp.dest('build/img'));
+function styles() {
+  return gulp.src(paths.styles.src)
+    .pipe(less())
+    .pipe(cssmin())
+    .pipe(rename({
+          basename: 'main',
+          suffix: '.min'
+        }))
+    .pipe(gulp.dest(paths.styles.dest));
 }
 
-// Minify and copy all JavaScript (except vendor scripts)
-// with sourcemaps all the way down
 function scripts() {
-  return gulp.src(paths.scripts)
-    .pipe(sourcemaps.init())
-      .pipe(coffee())
-      .pipe(uglify())
-      .pipe(concat('all.min.js'))
-    .pipe(sourcemaps.write())
-    .pipe(gulp.dest('build/js'));
+  return gulp.src(paths.scripts.src)
```

Comment:
So, this would be the final result?

```js
function scripts() {
  return gulp.src(paths.scripts, { sourcemaps: true })
    .pipe(coffee())
    .pipe(uglify())
    .pipe(concat('all.min.js'))
     // don't know if this is still needed
     .pipe(sourcemaps.write())
    .pipe(gulp.dest('build/js'));
}
```